py/pysparkling/context.py

```diff
@@ -119,12 +119,20 @@ def getOrCreate(spark_context, conf = None):
         # Create H2O REST API client
         h2o.connect(ip=h2o_context._client_ip, port=h2o_context._client_port)
         h2o_context.is_initialized = True
+        # Stop h2o when running standalone pysparkling scripts and the user does not explicitly close h2o
+        atexit.register(lambda: h2o_context.stop())
         return h2o_context
 
     def stop(self):
-        warnings.warn("H2OContext stopping is not yet fully supported...")
+        h2o.cluster().shutdown()
```

Comment:
It will shutdown JVM directly.

My point is that client (=PySparkling) should be to some extend stupid. Backend should be clever one to know how to properly shutdown cluster.